repairnator/src/main/java/fr/inria/spirals/repairnator/process/ProjectScanner.java

```diff
@@ -254,23 +235,43 @@ private boolean testBuild(Build build, boolean targetFailing) {
         return false;
     }
 
-    private List<Build> getListOfBuildsFromRepo(List<Repository> repos, boolean targetFailing) {
-        List<Build> result = new ArrayList<Build>();
+    private List<BuildToBeInspected> getListOfBuildsFromRepo(List<Repository> repos, RepairMode mode, boolean targetFailing) {
+        List<BuildToBeInspected> buildsToBeInspected = new ArrayList<BuildToBeInspected>();
 
-        this.buildsId = new ArrayList<Integer>();
         for (Repository repo : repos) {
             List<Build> repoBuilds = BuildHelper.getBuildsFromRepositoryWithLimitDate(repo, this.limitDate);
 
             for (Build build : repoBuilds) {
                 this.totalScannedBuilds++;
                 if (testBuild(build, targetFailing)) {
-                    result.add(build);
-                    this.buildsId.add(build.getId());
+                    if (mode == RepairMode.FORBEARS) {
+                        // First of all, here it is checked if the current passing build has a
+                        // previous build---if doesn't, nothing is made as this build is not
+                        // useful for Bears
+                        Build previousBuild = BuildHelper.getLastBuildOfSameBranchOfStatusBeforeBuild(build, null);
+                        if (previousBuild != null) {
+                            this.logger.debug("Build: " + build.getId());
+                            this.logger.debug("Previous build: " + previousBuild.getId());
+
+                            if (previousBuild.getBuildStatus() == BuildStatus.FAILED) {
+                                BuildToBeInspected buildToBeInspected = new BuildToBeInspected(build, previousBuild, ScannedBuildStatus.ONLY_FAIL);
```

Comment:
Don't agree: it's not "ONLY FAIL" here: you just test the "current" build and its previous one, so it's another status like "PASSING_AND_FAIL" or so.